libr/core/disasm.c

```diff
@@ -869,14 +870,15 @@ static int var_comparator(const RAnalVar *a, const RAnalVar *b){
 	return false;
 }
 
-static void handle_show_functions(RCore *core, RDisasmState *ds) {
+static void ds_show_functions(RDisasmState *ds) {
 	RAnalFunction *f;
+	RCore *core = ds->core;
 	bool demangle, call;
 	const char *lang;
 	char *fcn_name;
 	char *sign;
 
-	if (!core || !ds || !ds->show_functions) {
+	if (!ds->show_functions) {
```

Comment:
Plz corect me if i am wrong but this checks should belong to public api (aka r_core_* functions) not to internal ones.

BTW: They was only on this function nothing else. It is smth special about this function i am missing?